sound/soc/intel/common/soc-acpi-intel-cht-match.c

```diff
@@ -174,6 +174,13 @@ struct snd_soc_acpi_mach  snd_soc_acpi_intel_cherrytrail_machines[] = {
 		.sof_fw_filename = "sof-cht.ri",
 		.sof_tplg_filename = "sof-cht-cx2072x.tplg",
 	},
+	{
+		.id = "104C5122",
+		.drv_name = "sof_pcm512x",
```

Comment:
@plbossart with the introduction of default_fw_filename in the machine desc in SOF, we dont actually need the sof_fw_filename field in the acpi_mach anymore. Should we consider removing it?

Reply:
Oh btw Im asking for the future, it doesnt affect this PR.

Reply:
@ranj063 I don't mind if we use the default firmware if the machine driver does not provide a firmware file name, but I'd really like to keep the ability to use a different firmware for a specific platform, e.g. to allow for experiments before everything is merged.
in other works, ok to use a default but not ok to only use a default.

Reply:
@plbossart OK makes sense then in that case I guess we do need a minor change in the SOF driver to use the filename provided in the ACPI match table if provided to make use of it. I'll send a PR

Reply:
@plbossart I just realised something while working on the multi-client implementation. With multiple clients with separate machine drivers, does it make sense to have the fw_filename specified in each of them individually? Also, we will be probing the SOF DSP device first before we even probe the machine driver and the FW will be loaded when the SOF device is probed.

Reply:
well we need to have a single firmware, but each client might have a subset of the topologies.

Reply:
OK thats what I hae now. So then the only fw file that will be used is the default_fw_filename in the sof dev desc.